helmfile.d/values/prometheus-alerts-runbook-urls.yaml.gotmpl

```
@@ -0,0 +1,1101 @@
+runbookUrls:
+  {{ $prometheusOperatorRunbookUrl := "https://runbooks.prometheus-operator.dev/runbooks" }}
+  {{ $missingRunbook := "Missing runbook" }}
+  alertmanager:
+    {{ $runbookUrl := printf "%s/alertmanager/alertmanagerconfiginconsistent" $prometheusOperatorRunbookUrl }}
+    {{ $group := dig "alertmanager" "group" "" .Values.alerts.runbookUrls }}
+    {{ $override := dig "alertmanager" "AlertmanagerConfigInconsistent" $group .Values.alerts.runbookUrls }}
+    {{ if not (eq $override "") }}{{ $runbookUrl = $override }}{{ end }}
+    AlertmanagerConfigInconsistent: {{ $runbookUrl }}
+    {{ $runbookUrl := printf "%s/alertmanager/alertmanagerfailedreload" $prometheusOperatorRunbookUrl }}
+    {{ $group := dig "alertmanager" "group" "" .Values.alerts.runbookUrls }}
+    {{ $override := dig "alertmanager" "AlertmanagerFailedReload" $group .Values.alerts.runbookUrls }}
+    {{ if not (eq $override "") }}{{ $runbookUrl = $override }}{{ end }}
+    AlertmanagerFailedReload: {{ $runbookUrl }}
+    {{ $runbookUrl := printf "%s/alertmanager/alertmanagermembersinconsistent" $prometheusOperatorRunbookUrl }}
+    {{ $group := dig "alertmanager" "group" "" .Values.alerts.runbookUrls }}
+    {{ $override := dig "alertmanager" "AlertmanagerMembersInconsistent" $group .Values.alerts.runbookUrls }}
+    {{ if not (eq $override "") }}{{ $runbookUrl = $override }}{{ end }}
+    AlertmanagerMembersInconsistent: {{ $runbookUrl }}
```

Could we do something like this?

```suggestion
{{- define "tpl.prometheus-operator" -}}
{{- $output := lower .name | printf "https://runbooks.prometheus-operator.dev/runbooks/%s/%s" .group }}
{{- $output = .config | dig "group" $output }}
{{- $output = .config | dig .name $output }}
    {{ .name }}: {{ $output }}
{{- end }}

runbookUrls:
  alertmanager:
    {{- dict "name" "AlertmanagerConfigInconsistent" "group" "alertmanager" "config" .alertmanager | include "tpl.prometheus-operator" }}
    {{- dict "name" "AlertmanagerFailedReload" "group" "alertmanager" "config" .alertmanager | include "tpl.prometheus-operator" }}
    {{- dict "name" "AlertmanagerMembersInconsistent" "group" "alertmanager" "config" .alertmanager | include "tpl.prometheus-operator" }}
```

And do similar for other upstreams?

---

That can work, I'll look into it

---

Problem is that it doesn't work in cases where our alert does not have the same name as the path in the URL.

Should I set a default config value override in those cases?

---

Do we have that for alerts from prometheus-operator?

---

We didn't initially but I matched [some of these](https://github.com/elastisys/compliantkubernetes-apps/pull/2438/files#diff-58f7c4cca3233f66ab43830e39a0a221d49fada05f83c7a1d80ee737a56557bcR40) to fitting upstream runbooks, but I guess I can remove that and just put "Missing runbook".

---

We could always pass another one in then:
```yaml
{{- define "tpl.prometheus-operator" -}}
{{- $output := . | dig "override" .name }} 
{{- $output = lower $output | printf "https://runbooks.prometheus-operator.dev/runbooks/%s/%s" .group }}
{{- $output = .config | dig "group" $output }}
{{- $output = .config | dig .name $output }}
    {{ .name }}: {{ $output }}
{{- end -}}
{{-/* This with is just to simplify access to it */-}}
{{- with .Values.alerts.runbookUrls -}}
runbookUrls:
  alertmanager:
    {{- dict "name" "AlertmanagerConfigInconsistent" "group" "alertmanager" "config" .alertmanager | include "tpl.prometheus-operator" }}
    {{- dict "name" "AlertmanagerFailedReload" "group" "alertmanager" "config" .alertmanager | include "tpl.prometheus-operator" }}
    {{- dict "name" "AlertmanagerMembersInconsistent" "group" "alertmanager" override "differentrunbookname" "config" .alertmanager | include "tpl.prometheus-operator" }}
```

---

sure, though I found the "group" key in the dict to be confusing, so I'm using "subpath" instead